server/index.js

```diff
@@ -58,6 +58,92 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    res.json({});
+  } else {
+    const paramDb = [req.session.cartId];
+    const sql = `
+      select "c"."cartItemId",
+             "c"."price",
+             "p"."productId",
+             "p"."image",
+             "p"."name",
+             "p"."shortDescription"
+        from "cartItems" as "c"
+        join "products" as "p" using ("productId")
+       where "c"."cartId" = $1
+    `;
+
+    db.query(sql, paramDb).then(result => res.json(result.rows[0]));
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = parseInt(req.body.productId);
+  if (isNaN(req.body.productId) || productId < 0) {
+    return res.status(400).json({ error: 'Your requested productId is invalid.' });
+  }
+
+  const paramDb = [productId];
+  const sql = `
+    select "price"
+      from "products"
+     where "productId" = $1
+  `;
+
+  db.query(sql, paramDb)
+    .then(result => {
+      if (result.rows[0] === undefined) {
+        next(new ClientError('Requested productId may not exist in the database. Check your data agin.', 400));
```

Comment:
You should `throw` the error and not call `next()` to prevent the rest of the promise chain from executing.